pkg/chains/chain_filters.go

```diff
@@ -0,0 +1,60 @@
+package chains
+
+// ChainFilter is a function that filters chains based on some criteria
+type ChainFilter func(c Chain) bool
+
+// FilterExternalChains filters chains that are external
+func FilterExternalChains(c Chain) bool {
+	return c.IsExternal
+}
+
+// FilterGatewayObserver filters chains that are gateway observers
+func FilterGatewayObserver(c Chain) bool {
+	return c.CctxGateway == CCTXGateway_observers
+}
+
+// FilterConsensusEthereum filters chains that have the ethereum consensus
+func FilterConsensusEthereum(c Chain) bool {
+	return c.Consensus == Consensus_ethereum
+}
+
+// FilterConsensusBitcoin filters chains that have the bitcoin consensus
+func FilterConsensusBitcoin(c Chain) bool { return c.Consensus == Consensus_bitcoin }
+
+// FilterConsensusSolana filters chains that have the solana consensus
+func FilterConsensusSolana(c Chain) bool { return c.Consensus == Consensus_solana_consensus }
+
+// FilterChains applies a list of filters to a list of chains
+func FilterChains(chainList []Chain, filters ...ChainFilter) []Chain {
+	// Apply each filter to the list of supported chains
+	for _, filter := range filters {
+		var filteredChains []Chain
+		for _, chain := range chainList {
+			if filter(chain) {
+				filteredChains = append(filteredChains, chain)
+			}
+		}
+		chainList = filteredChains
+	}
+
+	// Return the filtered list of chains
+	return chainList
+}
+
+// CombineFilterChains combines multiple lists of chains into a single list
+func CombineFilterChains(chainLists ...[]Chain) []Chain {
```

Comment:
Why do we need to dedup chains here?

Reply:
why would we need duplicates ?
I don't see any harm in deduplicating

Reply:
Naming got me confused 😅 We can implement this as a generic `slices.Unique([]E)` under `pkg/slices`. Otherwise please name smth like `UniqueChains(list []Chain) []Chain`

Reply:
Its combines the output of multiple filters , and the naming is okay for now since its in the chain_filters file . Happy to implement a generic function though , I think that makes sense , ill add a issue for it .